src/training/optim.py

```diff
@@ -0,0 +1,133 @@
+from torch import optim
+
+def get_num_layer_for_transformer(var_name, num_max_layer):
+    first_layer_var_names = [
+        "visual.cls_token", 
+        "visual.mask_token", 
+        "visual.pos_embed", 
+        "visual.positional_embedding", 
+        "visual.patch_embed",
+        "visual.conv1", # name of patch embed in VisionTransformer
+        "text.pos_embed", 
+        "text.positional_embedding",
+        "text.token_embedding",
+        "text.transformer.embeddings.word_embeddings",
+        "text.transformer.embeddings.position_embeddings",
+        "text.transformer.embeddings.token_type_embeddings"
```

Comment:
will this work with all architectures we support? E.g. text models from HF

Reply:
@gabrielilharco no that's something else that needs to be addressed , as implemented it only works for built-in vit + text transformer, needs to at least detect if it will work and warn that it cannot be used for other models...

It'd be very useful for pretrained timm and HF text models, timm has functions that can calculate the layer decay but needs to be called if a timm model is used (that can be a diff PR), and not sure if HF has any built-in support to calculate layer-decay (discriminative LR) in a general way...